src/sentry/workflow_engine/endpoints/validators/metric_alert_detector.py

```diff
@@ -38,3 +43,75 @@ def validate(self, attrs):
         if len(conditions) > 2:
             raise serializers.ValidationError("Too many conditions")
         return attrs
+
+    def update_data_conditions(self, instance, data_conditions):
+        """
+        Update the data condition if it already exists, create one if it does not
+        """
+        if instance.workflow_condition_group:
+            try:
+                data_condition_group = DataConditionGroup.objects.get(
+                    id=instance.workflow_condition_group.id
+                )
+            except DataConditionGroup.DoesNotExist:
+                raise serializers.ValidationError("DataConditionGroup not found, can't update")
+        # TODO make one if it doesn't exist and data is passed?
+
+        for data_condition in data_conditions:
```

Comment:
would it be better to collect all the data conditions for the data condition group, delete them if they exist, and create new data conditions? we could use `bulk_delete` and `bulk_create` that way instead of one by one (fewer db hits) 🤔

Reply:
Hmm I was following the pattern we have now for alert rule triggers, I don't honestly know which is better. If there are say 10 data conditions but the update is only changing one of them, then we'd be unnecessarily deleting and re-creating 9 of them. I guess the way I have it we're always calling `update` on them even if there are no changes 🤔

Reply:
i also realized both `update()` and `bulk_create()` don't call `save()` which would send the pre-save signal :/ so really not sure which is better, we should try to validate them too

Reply:
The data does get validated, but I might be confused by your meaning here

Reply:
we have pre-save signals on DataConditions, Detectors, and Workflows to validate the json schemas